tests/components/broadlink/test_remote.py

```diff
@@ -0,0 +1,120 @@
+"""Tests for Broadlink remotes."""
+from base64 import b64decode
+
+from homeassistant.components.broadlink.const import DOMAIN, REMOTE_DOMAIN
+from homeassistant.components.remote import (
+    SERVICE_SEND_COMMAND,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.helpers.entity_registry import async_entries_for_device
+
+from . import get_device
+
+from tests.async_mock import call
+from tests.common import mock_device_registry, mock_registry
+
+REMOTE_DEVICES = ["Entrance", "Living Room", "Office", "Garage"]
+
+IR_PACKET = (
+    "JgBGAJKVETkRORA6ERQRFBEUERQRFBE5ETkQOhAVEBUQFREUEBUQ"
+    "OhEUERQRORE5EBURFBA6EBUQOhE5EBUQFRA6EDoRFBEADQUAAA=="
+)
+
+
+async def test_remote_setup_works(hass):
+    """Test a successful setup with all remotes."""
+    for device in map(get_device, REMOTE_DEVICES):
+        device_registry = mock_device_registry(hass)
+        entity_registry = mock_registry(hass)
+        mock_api, mock_entry = await device.setup_entry(hass)
+
+        device_entry = device_registry.async_get_device(
+            {(DOMAIN, mock_entry.unique_id)}, set()
+        )
+        entries = async_entries_for_device(entity_registry, device_entry.id)
+        remotes = {entry for entry in entries if entry.domain == REMOTE_DOMAIN}
+        assert len(remotes) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6103615)